kernel/src/engine/arrow_expression/tests.rs

```diff
@@ -157,6 +189,23 @@ fn test_literal_complex_type_array() {
     assert!(expected_values
         .zip(list_values.as_primitive::<Int32Type>())
         .all(|(a, b)| a == b));
+
+    let map_values = struct_values.column(3);
+    let map_array = map_values.as_map();
+    assert_eq!(map_array.keys().len(), 5 * 2 * 2);
+    assert_eq!(map_array.values().len(), 5 * 2 * 2); // I would think this should be 5 * 2 * 1
```

Comment:
cc @scovich - I was playing around with extending your complex types test and ran across a mismatch (of my understanding)